src/ray/core_worker/core_worker.cc

```diff
@@ -3120,6 +3121,7 @@ void CoreWorker::HandlePushTask(rpc::PushTaskRequest request,
           if (IsExiting()) {
             RAY_LOG(INFO) << "Queued task " << func_name
                           << " won't be executed because the worker already exited.";
+            send_reply_callback(Status::Invalid("Worker has exited."), nullptr, nullptr);
```

Comment:
Not 100% sure about this. It seems HandleTask eventually should call send_reply_callback?

Reply:
yeah HandleTask should reply after task is finished.